applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -108,6 +121,73 @@ static struct bt_audio_discover_params audio_source_discover_param[CONFIG_BT_MAX
 static int discover_source(struct bt_conn *conn);
 #endif /* CONFIG_STREAM_BIDIRECTIONAL */
 
+/**
+ * @brief  Get the common presentation delay for all headsets or
+ *         set to 0 on an error
+ *
+ * @param  mode         The mode of search for the presentation delay
+ * @param  index        The index of the headset to test against
+ * @param  pres_dly_us  A pointer to the presentation delay found
+ *
+ * @return 0 if succesfull, -EINVAL on an error
+ */
+static int headset_pres_delay_find(enum le_audio_pres_search mode, uint8_t index,
+				   uint32_t *pres_dly_us)
+{
+	uint32_t pres_dly_min = headsets[index].sink_ep->qos_pref.pd_min;
+	uint32_t pres_dly_max = headsets[index].sink_ep->qos_pref.pd_max;
+	uint32_t pref_dly_min = headsets[index].sink_ep->qos_pref.pref_pd_min;
+	uint32_t pref_dly_max = headsets[index].sink_ep->qos_pref.pref_pd_max;
+
+	for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
+		if (headsets[i].sink_ep != NULL) {
+			pres_dly_min = MAX(pres_dly_min, headsets[i].sink_ep->qos_pref.pd_min);
+			pres_dly_max = MIN(pres_dly_max, headsets[i].sink_ep->qos_pref.pd_max);
+			pref_dly_min = MAX(pref_dly_min, headsets[i].sink_ep->qos_pref.pref_pd_min);
+			pref_dly_max = MIN(pref_dly_max, headsets[i].sink_ep->qos_pref.pref_pd_max);
+		}
+	}
+
+	*pres_dly_us = 0;
+
+	switch (mode) {
+	case AUDIO_PRES_DELAY_MODE_MIN:
+		*pres_dly_us = pres_dly_min;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_MAX:
+		*pres_dly_us = pres_dly_max;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_PREF_MIN:
+		*pres_dly_us = pref_dly_min;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_PREF_MAX:
+		*pres_dly_us = pref_dly_max;
+		break;
+
+	case AUDIO_PRES_DELAY_MODE_LOCAL:
+		if (IN_RANGE(CONFIG_BT_AUDIO_LOCAL_PRESENTATION_DELAY_US, pres_dly_min,
+			     pres_dly_max)) {
+			*pres_dly_us = CONFIG_BT_AUDIO_LOCAL_PRESENTATION_DELAY_US;
+		} else {
+			LOG_ERR("Preferred local presentaion delay outside of range");
+			return -EINVAL;
```

Comment:
I think we need to discuss what to do here. My thought is that it should revert to the closest of the four values above, instead of returning an error that would reject the connection.